Lib/test/test_tstring.py

```diff
@@ -0,0 +1,441 @@
+        self.assertEqual(combined.interpolations[0].value, name)
+        self.assertEqual(combined.interpolations[0].expression, "name")
+        self.assertEqual(combined.interpolations[0].conversion, None)
+        self.assertEqual(combined.interpolations[0].format_spec, "")
+        self.assertEqual(f(combined), "Hello, Python")
+
+        # Test string + template
+        t = "Hello, " + t"{name}"
+        self.assertEqual(t.strings, ("Hello, ", ""))
+        self.assertEqual(t.interpolations[0].value, name)
+        self.assertEqual(t.interpolations[0].expression, "name")
+        self.assertEqual(t.interpolations[0].conversion, None)
+        self.assertEqual(t.interpolations[0].format_spec, "")
+        self.assertEqual(f(t), "Hello, Python")
+
+    def test_nested_templates(self):
+        # Test a template inside another template expression
+        name = "Python"
+        inner = t"{name}"
+        t = t"Language: {inner}"
+
+        self.assertEqual(t.strings, ("Language: ", ""))
+        self.assertEqual(t.interpolations[0].value.strings, ("", ""))
+        self.assertEqual(t.interpolations[0].value.interpolations[0].value, name)
+        self.assertEqual(t.interpolations[0].value.interpolations[0].expression, "name")
+        self.assertEqual(t.interpolations[0].value.interpolations[0].conversion, None)
+        self.assertEqual(t.interpolations[0].value.interpolations[0].format_spec, "")
+        self.assertEqual(t.interpolations[0].expression, "inner")
+        self.assertEqual(t.interpolations[0].conversion, None)
+        self.assertEqual(t.interpolations[0].format_spec, "")
+
+    def test_ast_structure(self):
+        # Test AST structure for simple t-string
+        tree = ast.parse('t"Hello"')
+        self.assertIsInstance(tree.body[0].value, ast.TemplateStr)
+        self.assertIsInstance(tree.body[0].value.values[0], ast.Constant)
+
+        # Test AST for t-string with interpolation
+        tree = ast.parse('t"Hello {name}"')
+        self.assertIsInstance(tree.body[0].value, ast.TemplateStr)
+        self.assertIsInstance(tree.body[0].value.values[0], ast.Constant)
+        self.assertIsInstance(tree.body[0].value.values[1], ast.Interpolation)
+
+    def test_error_conditions(self):
+        # Test syntax errors
+        with self.assertRaisesRegex(SyntaxError, "'{' was never closed"):
+            eval("t'{")
+
+        with self.assertRaisesRegex(SyntaxError, "t-string: expecting '}'"):
+            eval("t'{a'")
+
+        with self.assertRaisesRegex(SyntaxError, "t-string: single '}' is not allowed"):
+            eval("t'}'")
+
+        # Test missing variables
+        with self.assertRaises(NameError):
+            eval("t'Hello, {name}'")
+
+        # Test invalid conversion
+        num = 1
+        with self.assertRaises(SyntaxError):
+            eval("t'{num!z}'")
+
+    def test_literal_concatenation(self):
+        # Test concatenation of t-string literals
```

Comment:
What if you concatenate a t-string with another kind of string (f-string, raw string, byte string)? Those should also be tested.